base/strings/utf8proc.jl

```diff
@@ -74,8 +74,12 @@ function utf8proc_map(s::String, flags::Integer)
                    (Ptr{UInt8}, Cssize_t, Ref{Ptr{UInt8}}, Cint),
                    s, sizeof(s), p, flags)
     result < 0 && error(unsafe_string(ccall(:utf8proc_errmsg, Cstring,
-                                         (Cssize_t,), result)))
-    unsafe_wrap(String, p[], result, true)::String
+                                            (Cssize_t,), result)))
+    ptr = p[]
+    # TODO: avoid copy
```

Comment:
For example, here is code to do `utf8proc_map` on a Julia-supplied buffer in Julia 0.5:
```jl
utf8proc_error(result) = error(unsafe_string(ccall(:utf8proc_errmsg, Cstring, (Cssize_t,), result)))

function utf8proc_map(str::String, options)
    nwords = ccall(:utf8proc_decompose, Int, (Ptr{UInt8}, Int, Ptr{UInt8}, Int, Cint),
                   str, sizeof(str), C_NULL, 0, options)
    nwords < 0 && utf8proc_error(nwords)
    buffer = Vector{UInt8}(nwords*4)
    nwords = ccall(:utf8proc_decompose, Int, (Ptr{UInt8}, Int, Ptr{UInt8}, Int, Cint),
                   str, sizeof(str), buffer, nwords, options)
    nwords < 0 && utf8proc_error(nwords)
    nbytes = ccall(:utf8proc_reencode, Int, (Ptr{UInt8}, Int, Cint), buffer, nwords, options)
    nbytes < 0 && utf8proc_error(nbytes)
    return String(resize!(buffer, nbytes))
end

julia> utf8proc_map("Foo", Base.UTF8proc.UTF8PROC_CASEFOLD)
"foo"
```